Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -18,13 +19,100 @@
+                        AZStd::string potentialStreamingImagePath(assetHint);
+                        AZ::StringFunc::Replace(potentialStreamingImagePath, ".gradimage", "");
+                        potentialStreamingImagePath += imageExtension + ".streamingimage";
+
+                        // Check if there is a valid streaming image asset for this path
+                        AZ::Data::AssetCatalogRequestBus::BroadcastResult(fixedAssetId, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetIdByPath, potentialStreamingImagePath.c_str(), azrtti_typeid<AZ::Data::Asset<AZ::RPI::StreamingImageAsset>>(), false);
+                        if (fixedAssetId.IsValid())
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+
+            // Replace the old gradimage with new AssetId for streaming image asset (if we needed to replace)
+            if (fixedAssetId.IsValid())
+            {
+                configInstance->m_imageAsset = AZ::Data::AssetManager::Instance().GetAsset<AZ::RPI::StreamingImageAsset>(fixedAssetId, AZ::Data::AssetLoadBehavior::QueueLoad);
+            }
+            // Otherwise, this was already a streaming image asset, so just load it like normal
+            else
+            {
+                result.Combine(ContinueLoadingFromJsonObjectField(
+                    &configInstance->m_imageAsset, azrtti_typeid<decltype(configInstance->m_imageAsset)>(), inputValue, "ImageAsset", context));
+            }
+        }
+
+        result.Combine(ContinueLoadingFromJsonObjectField(
+            &configInstance->m_tilingX, azrtti_typeid<decltype(configInstance->m_tilingX)>(), inputValue, "TilingX", context));
+
+        result.Combine(ContinueLoadingFromJsonObjectField(
+            &configInstance->m_tilingY, azrtti_typeid<decltype(configInstance->m_tilingY)>(), inputValue, "TilingY", context));
+
+        return context.Report(result,
+            result.GetProcessing() != JSR::Processing::Halted ?
+            "Successfully loaded ImageGradientConfig information." :
+            "Failed to load ImageGradientConfig information.");
+    }
+
+    AZ_CLASS_ALLOCATOR_IMPL(JsonImageGradientConfigSerializer, AZ::SystemAllocator, 0);
+
     void ImageGradientConfig::Reflect(AZ::ReflectContext* context)
     {
+        if (auto jsonContext = azrtti_cast<AZ::JsonRegistrationContext*>(context))
+        {
+            jsonContext->Serializer<JsonImageGradientConfigSerializer>()->HandlesType<ImageGradientConfig>();
+        }
+
         AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context);
         if (serialize)
         {
             serialize->Class<ImageGradientConfig, AZ::ComponentConfig>()
-                ->Version(1)
+                ->Version(2)
                 ->Field("ImageAsset", &ImageGradientConfig::m_imageAsset)
```

Comment:
Suggestion: If you rename the field to "StreamingImageAsset" or something, it will be trivial to detect the new vs old version above, instead of having to go further into the type inspection.

Reply:
Good idea, with that I can early-out much quicker if we are in version 2, will update!